File: GO/Lei_de_Ohm.go
package main

import "fmt"

func PdadoVI(V float32, I float32) float32{
	return V*I;
}

func IdadoVR(V float32, R float32) float32{
	return V/R;
}

func RdadoVI(V float32, I float32) float32{
	return V/I;
}

func RdadoVP(V float32, P float32) float32{
	return (V*V)/P;
}

func main(){
	var(
		opcao		int
		V, I, P, R	float32
	)
	
	for{
		fmt.Println("\n1- Digite 1 para calcular P dados V e I")
		fmt.Println("2- Digite 2 para calcular I dados V e R")
		fmt.Println("3- Digite 3 para calcular R dados P e V")
		fmt.Println("4- Digite 4 para calcular R dados V e I")
		fmt.Println("5- Digite 5 para SAIR\n")
		fmt.Scanf("%d", &opcao)
			
		switch opcao{
			case 1:
				fmt.Print("\nDigite V: ")
				fmt.Scanf("%f", &V)
				fmt.Print("Digite I: ")
				fmt.Scanf("%f", &I)
				P = PdadoVI(V, I)
				fmt.Println("\nO valor de P é: ", P)
				break
	
			case 2:
				fmt.Print("\nDigite V: ")
				fmt.Scanf("%f", &V)
				fmt.Print("Digite R: ")
				fmt.Scanf("%f", &R)
				I = IdadoVR(V, R)
				fmt.Println("\nO valor de I é: ", I)
				break
			
			case 3:
				fmt.Print("\nDigite V: ")
				fmt.Scanf("%f", &V)
				fmt.Print("Digite P: ")
				fmt.Scanf("%f", &P)
				R = RdadoVP(V, P)
				fmt.Println("\nO valor de R é: ", R)
				break
			
			case 4:
				fmt.Print("\nDigite V: ")
				fmt.Scanf("%f", &V)
				fmt.Print("Digite I: ")
				fmt.Scanf("%f", &I)
				R = RdadoVI(V, I)
				fmt.Println("\nO valor de R é: ", R)
				break
			
			case 5:
				fmt.Println("\nEncerrando a aplicação...")
				return
			
			default:
				fmt.Println("\nOpção inválida! \n")
					
		}
	}
}

File: Java/Lei_de_Ohm.java
import java.util.Scanner;

public class Lei_de_Ohm {

	public static void main(String[] args) {
		int opcao = 0;
		float V = 0, I = 0, R = 0, P = 0;
		
		Scanner ler = new Scanner(System.in);

		while(true){
			
			System.out.println("\n1- Digite 1 para calcular P dados V e I");
			System.out.println("2- Digite 2 para calcular I dados V e R");
			System.out.println("3- Digite 3 para calcular R dados P e V");
			System.out.println("4- Digite 4 para calcular R dados V e I");
			System.out.println("5- Digite 5 para SAIR\n\n");
			opcao = ler.nextInt();
			
			switch(opcao){
				case 1:
					System.out.print("\nDigite V: ");
					V = ler.nextFloat();
					System.out.print("Digite I: ");
					I = ler.nextFloat();
					P = PdadoVI(V, I);
					System.out.print("O valor de P é: " + P +"\n");
					break;
					
				case 2:
					System.out.print("\nDigite V: ");
					V = ler.nextFloat();
					System.out.print("Digite R: ");
					R = ler.nextFloat();
					I = IdadoVR(V, R);
					System.out.print("O valor de I é: " + I +"\n");
					break;
					
				case 3:
					System.out.print("\nDigite V: ");
					V = ler.nextFloat();
					System.out.print("Digite P: ");
					P = ler.nextFloat();
					R = RdadoVP(V, P);
					System.out.print("O valor de R é: " + R +"\n");
					break;
					
				case 4:
					System.out.print("\nDigite V: ");
					V = ler.nextFloat();
					System.out.print("Digite I: ");
					I = ler.nextFloat();
					R = RdadoVI(V, I);
					System.out.print("O valor de R é: " + R +"\n");
					break;
				
				case 5: 
					System.out.println("\nEncerrando a aplicação...");
					return;
					
					
				default:
					System.out.print("\nOpção inválida! \n\n" );
					break;
			}
		}
	}
	
	public static float PdadoVI( float V , float I){
		return V*I;
	}
	
	public static float IdadoVR( float V , float R){
		return V/R;
	}
	
	public static float RdadoVP( float V , float P){
		return (V*V)/P;
	}
	
	public static float RdadoVI( float V , float I){
		return V/I;
	}	
}
